src/main/java/frc/robot/subsystems/swerve/SwerveModule.java

```diff
@@ -47,44 +48,62 @@ public SwerveModule(int moduleID) {
+                || cancoder.getBusVoltage() < 2);
 
-        steerMotor.configFactoryDefault();
-        steerMotor.setNeutralMode(NeutralMode.Brake);
-        steerMotor.config_kP(0, MODULE_STEER_KP);
-        steerMotor.config_kD(0, MODULE_STEER_KD);
-        ErrorCode error = steerMotor.setSelectedSensorPosition(
-                degreesToTicks(cancoder.getPosition() % 360));
-        if (error != ErrorCode.OK) {
-            throw new IllegalStateException("Swerve module " + id + " couldn't initialize");
-        }
+        initialize();
     }
 
-    private void configCancoder() {
+    public void initialize() {
         CANCoderConfiguration config = new CANCoderConfiguration();
         config.absoluteSensorRange = AbsoluteSensorRange.Unsigned_0_to_360;
         config.magnetOffsetDegrees = 0;
         config.sensorDirection = false;
         config.initializationStrategy = SensorInitializationStrategy.BootToAbsolutePosition;
         cancoder.configAllSettings(config);
-        cancoder.setPosition(cancoder.getAbsolutePosition() - CANCODER_OFFSETS[id]);
+        checkCtreError(cancoder.getLastError());
+        double absolute = cancoder.getAbsolutePosition();
+        checkCtreError(cancoder.getLastError());
+        cancoder.setPosition(absolute - CANCODER_OFFSETS[id]);
+        checkCtreError(cancoder.getLastError());
+
+        checkCtreError(driveMotor.configFactoryDefault());
+        driveMotor.setNeutralMode(NeutralMode.Brake);
+        checkCtreError(driveMotor.getLastError());
+        driveMotor.setInverted(true);
+        checkCtreError(driveMotor.getLastError());
+        checkCtreError(driveMotor.setSelectedSensorPosition(0D));
+        checkCtreError(driveMotor.config_kP(0, MODULE_DRIVE_KP));
+
+        checkCtreError(steerMotor.configFactoryDefault());
+        steerMotor.setNeutralMode(NeutralMode.Brake);
+        checkCtreError(steerMotor.getLastError());
+        checkCtreError(steerMotor.config_kP(0, MODULE_STEER_KP));
+        checkCtreError(steerMotor.config_kD(0, MODULE_STEER_KD));
+        double position = cancoder.getPosition() % 360;
+        checkCtreError(cancoder.getLastError());
+        checkCtreError(steerMotor.setSelectedSensorPosition(degreesToTicks(position)));
+    }
+
+    public void log() {
+        SmartDashboard.putNumber("Module " + id + " cancoder", cancoder.getPosition());
+        SmartDashboard.putNumber("Module " + id + " cancoder absolute", cancoder.getAbsolutePosition());
+        SmartDashboard.putNumber("Module " + id + " cancoder offset",
+                cancoder.getPosition() - cancoder.getAbsolutePosition());
+        SmartDashboard.putNumber("Module " + id + " angle",
+                ticksToDegrees(steerMotor.getSelectedSensorPosition()));
+    }
+
+    private void checkCtreError(ErrorCode error) {
+        if (error != ErrorCode.OK) {
+            throw new IllegalStateException("Swerve module " + id + " couldn't initialize");
```

Comment:
Since we are running this at teleopInit and during matches on a button press, we should consider not throwing an exception unless we are going to catch it.